City-of-Helsinki/drupal-helfi-platform-config
# UHF-8441 hearing external entity

# [UHF-8441](https://helsinkisolutionoffice.atlassian.net/browse/UHF-8441)

## What was done

Added external entity for hearings
Added paragraph to Landing page which shows hearing entities

## How to install
* Make sure your instance is up and running on latest dev branch.
    * `git pull origin dev`
    * `make fresh`
* Update the Helfi Platform config
    * `composer require drupal/helfi_platform_config:dev-UHF-8441_hearing_external_entity`
    * `composer require drupal/hdbt:dev-UHF-8442_hearing_paragraph`
* Run `drush en helfi_paragraphs_hearings`
* Run `make drush-updb drush-cr`
* Run `make shell`
* Inside the shell run `drush locale:check; drush locale:update; drush cr`

## How to test

- Log in as administrator
- Go and edit any landing page  
  - Add a hearing paragraph to the page (kuulemiset). Add title and description and save the landing page
- Go check the landing page content
- You should see the hearings-paragraph on the page. Paragraph may or may not contain latest open hearings.
- Compare the hearings to the list [found here](https://kerrokantasi.hel.fi/hearings/list?lang=fi)
  - The order should match 

Other things to test:
- Change language: If one or more of the hearings doesn't contain translation with currently selected language 
  - Content is rendered in other language
  - Lang-attribute is added to the content element

- If api returned no items: 
  - Go to `Hearings.php` 
  - Around line 123, edit the line with `'open' => 'true',`, change true to false
  - Run `drush cr`
  - Reload the page

- Test list without items:
  - Go to Hearings.php and add `return [];` on the first line of query-function
  - run `drush cr`
  - reload the page  
    

## Other PRs
[HDBT](https://github.com/City-of-Helsinki/drupal-hdbt/pull/712)


[UHF-8441]: https://helsinkisolutionoffice.atlassian.net/browse/UHF-8441?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
